manim/mobject/mobject.py

```diff
@@ -1343,21 +1344,81 @@ def __init__(self, *mobjects, **kwargs):
 
 
 class _AnimationBuilder:
-    def __init__(self, mobject, generate_target=True):
+    def __init__(self, mobject, overridden_animations, generate_target=True):
         self.mobject = mobject
+        self.overridden_animations = overridden_animations
         if generate_target:
             self.mobject.generate_target()
 
     def __getattr__(self, method_name):
         method = getattr(self.mobject.target, method_name)
 
+        if hasattr(method, "_animation_override"):
+            return MethodType(method._animation_override, self.mobject)
```

Comment:
Ths also isn't necessary anymore, right? And you shouldn't need to import `MethodType` anymore either

Reply:
Ah, yes. Without removing this, the actual implementation that is supposed to take over was never tested -- amusing.

Now, as I have actually tested this: `method._animation_override(*method_args, **method_kwargs)` does not work, because the original mobject ("`self`") is not passed to the method stored in `_animation_override` -- what do you think is the best solution? Should I just simply pass `self.mobject` as the first positional argument (`method._animation_override(self.mobject, *method_args, **method_kwargs)`)?

Reply:
Yeah, that seems best to me, and should be in line with standard behavior.

Reply:
Okay, that works. Now that I've actually tested this approach, I am not too happy about returning an `AnimationGroup`: because I currently always include the corresponding `_MethodAnimation` with it, there are unwanted side effects (because now the group contains multiple animations of the same mobject -- or, to be more precise, of a mobject and a submobject contained in it), which is not really great.

A workaround that comes to my mind right now is the following: an `_AnimationBuilder` could track whether or not a non-overridden method was passed at some point, and return an `AnimationGroup` which only contains the corresponding `_MethodAnimation` if necessary.

I'd also consider logging a warning if overridden and non-overridden methods are mixed, because I'm not sure whether this can work at all.

What do you think?

Reply:
If mixing overridden animations and normal `_MethodAnimation`s doesn't work, then I'd rather it just error if you try to mix them, whether that be with `obj.animate.overridden_method().normal_method()` or with `obj.animate.normal_method().overridden_method()`, or even `obj.animate.overridden_method1().overridden_method2()`. Either that or this whole animation override stuff shouldn't really exist and perhaps should be more explicit for the user, I'm not sure. Maybe @eulertour would have more insight with regards to multiple animations of the same mobject and chaining `.animate`

Reply:
That sounds fair.

For the record, I still (very much!) want a way to attach custom animations to method animations, but I'd be fine if overridden methods could not be chained (ideally even by providing an explicit error message).